rust-api/src/api.rs

```diff
@@ -53,36 +61,63 @@ impl Api {
 
     #[oai(path = "/seed/:user", method = "get")]
     async fn seed(&self, data_pool: Data<&PgPool>, user: Path<String>) -> PlainText<String> {
+        //let contents = fs::read_to_string(filename)
+        //    .expect("Something went wrong reading the file");
         // For each json file in /example_forms, add to db, for current user
         // insert into airport values (‘San Francisco’,’SFO’,ARRAY[23.42,-34.42, 23.34]);
 
-        new_form(
-            data_pool.0,
-            1, //TODO: Use the filename for the ID
-            "test user",
-            "key",
-            "display_name",
-            json!({"one": "two"})
-        ).await
-        .expect("Form not inserted into db");
 
+        seed_data_for_user(&user.0, data_pool.0)
+            .await
+            .expect("Seeding data failed for user");
 
         PlainText(format!("forms created for user: {}", user.0))
     }
 }
 
+async fn seed_data_for_user(user: &str, pool: &PgPool) -> Result<(), Error> {
+    let paths = fs::read_dir(&path::Path::new("./example_forms")).unwrap();
+
+    let file_names = paths.filter_map(|entry| {
+        entry.ok().and_then(|e| {
+            let file_name = e.path()
+                .file_name()
+                .and_then(|n| {
+                    n.to_str()
+                    .map(String::from)
+                });
+            file_name
+        })
+    }).collect::<Vec<String>>();
+
+    for form_file in file_names {
+        let path: PathBuf = ["./example_forms", &form_file].iter().collect();
+        let file_content = fs::read_to_string(path).unwrap();
+
+        new_form(
+            user,
+            &form_file,
+            &form_file,
+            json!(file_content),
+            pool,
+        ).await?;
+    }
+    Ok(())
+}
+
 async fn new_form(
-    pool: &PgPool,
-    id: i64,
     username: &str,
     key: &str,
     display_name: &str,
     form: Value,
-) -> Result<sqlx::postgres::PgRow, Error> {
+    pool: &PgPool,
+) -> Result<Form, Error> {
         sqlx::query_as!(
             Form, 
-            "INSERT INTO forms VALUES ($1, $2, $3, $4, $5);",
-            id,
+            "INSERT INTO forms
+             VALUES ($1, $2, $3, $4, $5)
+             RETURNING *;",
```

Comment:
you have to add `RETURNING` if you want postgres to give you the table back, otherwise you just get "1 row(s) updated" etc

Reply:
took me longer than I care to admit to realise this...